kernel/kernel-api/src/main/java/io/delta/kernel/internal/util/ColumnMapping.java

```diff
@@ -129,11 +132,22 @@ public static boolean isColumnMappingModeEnabled(ColumnMappingMode columnMapping
     return columnMappingMode == ColumnMappingMode.ID || columnMappingMode == ColumnMappingMode.NAME;
   }
 
-  public static Metadata updateColumnMappingMetadata(
+  /**
+   * Updates the column mapping metadata if needed based on the column mapping mode and whether the
+   * icebergCompatV2 is enabled. If column mapping/iceberg compat info is already present in the
+   * metadata, this method does nothing and returns an empty Optional. Callers can avoid updating
+   * the metadata if the metadata has not changed.
+   *
+   * @param metadata Current metadata.
+   * @param columnMappingMode New column mapping mode.
```

Comment:
nit - kinda weird that these are both args considering we get `columnMappingMode` from `metadata` when calling